install/kubernetes/cilium/templates/cilium-agent/validatingadmissionpolicy.yaml

```
@@ -0,0 +1,59 @@
+{{- if and .Values.validatingAdmissionPolicy.enabled (.Capabilities.APIVersions.Has "admissionregistration.k8s.io/v1alpha1/ValidatingAdmissionPolicy") }}
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "validate-cilium-network-policies"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   ["cilium.io"]
+        apiVersions: ["v2"]
+        operations:  ["CREATE", "UPDATE"]
+        resources:   ["ciliumnetworkpolicies", "ciliumclusterwidenetworkpolicies"]
+  validations:
+    - expression:
+        (
+          has(object.spec.ingress)
+          && object.spec.ingress.exists(x, has(x.toPorts)
```

Members of `ingress` do not need `toPorts`. My proposed change checks if `toPorts` does not exist and then proceeds to validate `toPorts` if it does.
```suggestion
          && object.spec.ingress.exists(x, !has(x.toPorts)
```